docs/src/developer/DEV_CONTAINER.md

```diff
@@ -1,86 +1,109 @@
+VSCode Dev Containers are great for developing as they contain all of the project's dependencies and in the end you have a ready-to-use dockerFile which can also be used in your CI/CD pipeline.
 
 It provides a normalized experience to all VSCode users across MacOS, Windows and Linux. All you need to get going is a the open source docker engine (not Docker Desktop!) and docker-compose.
 
 More information about Dev Containers can be found from the [official documentation](https://code.visualstudio.com/docs/devcontainers/containers).
 
-## Setting up Docker (OSS - Variant)
+## Pre-requisites
 
-### MacOS (amd64/arm64)
+Before VSCode Dev container can be used, you need to install a few things.
 
-There are different ways to install the docker engine on MacOS, however this guide will focus on using [colima](https://github.com/abiosoft/colima) which uses `limactl` under the hood to create a virtual machine and supporting functionality such as automatic port forwarding.
+1. Install `docker` and `docker-compose` if you do not already have these
 
-1. Install homebrew. You can also install homebrew without `sudo` rights using this [guide](https://docs.brew.sh/Installation#untar-anywhere-unsupported)
+    Checkout the [INSTALLING_DOCKER](./INSTALLING_DOCKER.md) instructions to help guide you
 
-    ✨ **Tip** ✨
+1. Install VSCode and make sure it is up to date
 
-    Make sure you read the homebrew instructions which are printed out on the console 😉
+2. Install the Remote Development extension
 
-2. Install [colima](https://github.com/abiosoft/colima) and the docker cli
+    ```
+    ms-vscode-remote.vscode-remote-extensionpack
+    ```
+
+## Getting started with Dev Containers
+
+Once the pre-requisites have been installed, you are ready to start using the dev container. You only need to run this instructions once. If you have already setup your dev container and just want to know how to re-open it, please read the next section.
+
+There are two main methods when setting up the dev container, we recommend using the `Method 1`. `Method ` uses the `Clone in Named Container Volume` strategy as  it is more performant on most Operating Systems. For more information about it please refer to the following [Dev Container documentation](https://code.visualstudio.com/remote/advancedcontainers/improve-performance). The only downside to this method is that the code will be stored in a docker volume. This means you can not directly navigate to the project from your the systems file explorer, however you can still `download` files from it from VSCode file explorer.
+
+But in both methods it is recommended to fork the project first, then clone from your fork. You can fork from the [GitHub](https://github.com/thin-edge/thin-edge.io) website, and then copy the git url from your fork.
+
```

Comment:
Also worth mentioning that it is **not** recommended to clone the repo inside the container directly, as it might lead to your work getting lost when the container is rebuilt in future.

Reply:
You can't actually clone it inside the container directly, as the container does not get built without having access to the code.

The `clone in named container volume` keeps a persistent volume which does not get destroyed when rebuilding the container. But generally there should be very low risk of loosing code because users should not be scared of doing commits and pushing them to the branch on their fork. This is a good habit to get into in general, as your computer can die at any point (or get lost, stolen etc.).